src/Viewer.tsx

```diff
@@ -17,58 +21,44 @@ export interface WidgetProps {
   width: number;
 }
 
-const agentColors = [
-  '#fee34d',
-  '#f7b232',
-  '#bf5736',
-  '#94a7fc',
-  '#ce8ec9',
-  '#58606c',
-  '#0ba345',
-  '#9267cb',
-  '#81dbe6',
-  '#bd7800',
-  '#bbbb99',
-  '#5b79f0',
-  '#89a500',
-  '#da8692',
-  '#418463',
-  '#9f516c',
-  '#00aabf',
-];
-
 function ViewerWidget(props: WidgetProps): JSX.Element {
+  const [modelInfo, setModelInfo] = useState<ModelInfo>({});
+
+  const handleTrajectoryData = (data: any) => {
+    setModelInfo(data.modelInfo);
+  };
+
```

Comment:
This is a better way of getting the `modelInfo`, we don't parse the string a second time in `widget.ts`